src/cgame/default/viewcontrollers/KeysViewController.c

```diff
@@ -104,8 +104,8 @@ static void loadView(ViewController *self) {
 
 			Cg_BindInput((View *) stackView, "Attack", "+attack");
 			Cg_BindInput((View *) stackView, "Grapple Hook", "+hook");
-			Cg_BindInput((View *) stackView, "Next weapon", "weapon_next");
-			Cg_BindInput((View *) stackView, "Previous weapon", "weapon_previous");
+			Cg_BindInput((View *) stackView, "Next weapon", "cg_weapon_next");
+			Cg_BindInput((View *) stackView, "Previous weapon", "cg_weapon_previous");
```

Comment:
Delay is still there because it makes switching to weapons faster, actually. The reason is simple: you don't want to switch between 6 different weapons to get the one you need. That's why there's a short delay between scrolling and when it actually switches. However, you can bypass this: clicking will force the weapon you're pointing at to switch to it immediately.